numpy/core/src/multiarray/dtypemeta.c

```diff
@@ -698,6 +698,28 @@ object_common_dtype(
     return cls;
 }
 
+int
+object_fill_zero_value(PyArrayObject *arr)
+{
+    PyObject *zero = PyLong_FromLong(0);
+    PyArray_FillObjectArray(arr, zero);
+    Py_DECREF(zero);
+    if (PyErr_Occurred()) {
+        return -1;
+    }
+    return 0;
+}
+
+int
+void_fill_zero_value(PyArrayObject *arr)
+{
+    if (PyDataType_REFCHK(PyArray_DESCR(arr))) {
+        if (object_fill_zero_value(arr) < 0) {
```

Comment:
OK, I guess here there is a reason to keep that `PyArray_FillObjectArray` thing around for this...